proposals/0383-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,179 @@
+
+Additionally, `let` wrapped properties add code safety where a user wants to expressly remove access to a property’s mutators after initializing the property once or simply does not need a mutable property wrapper. This could be useful for property wrappers that do not change or are reference types.
+
+## Proposed solution
+
+We propose to allow the application of property wrappers to let declared properties, which will permit the wrapper type to be initialized only once without affecting the implementation of the backing storage.
+
+For example, [TSPL](https://docs.swift.org/swift-book/LanguageGuide/Properties.html#ID617) defines a `SmallNumber` property wrapper and applies it to `UnitRectangle` properties:
+```
+@propertyWrapper
+struct SmallNumber {
+  private var maximum: Int
+  private var number: Int
+
+  var wrappedValue: Int {
+    get { return number }
+    set { number = min(newValue, maximum) }
+  }
+
+  init(wrappedValue: Int) {
+    maximum = 12
+    number = min(wrappedValue, maximum)
+  }
+}
+
+struct UnitRectangle {
+  @SmallNumber var height: Int = 1
+  @SmallNumber var width: Int = 1
+}
+```
+Initial values for `height` and `width` are set using the `init(wrappedValue:)` initializer of the `SmallNumber` property wrapper type. To ensure that `height` and `width` are not changed again, we could add logic to the `wrappedValue` setter to check if the property was already initialized and prevent re-assignment. However, this is an inconvenient solution.
+
+Instead, we could declare these properties with a `let`, synthesize a local `let` constant for the backing storage property (prefixed with an underscore), and only allow the property wrapper to be initialized once, passing the assigned value to `init(wrappedValue:)`. Now, rewriting `UnitRectangle`’s properties as `let` constants will translate to:
+```
+private let _height: SmallNumber = SmallNumber(wrappedValue: 1)
+var height: Int {
+  get { return _height.wrappedValue }
+}
+
+private let _weight: SmallNumber = SmallNumber(wrappedValue: 1)
+var weight: Int {
+  get { return _weight.wrappedValue }
+}
+```
+and results in code that is easy to write and understand:
+```
+struct UnitRectangle {
+  @SmallNumber let height: Int = 1
+  @SmallNumber let width: Int = 1
+}
+```
+
+Property wrappers with `let` declarations will be allowed both as members and local declarations, as envisioned by SE-0258 for `var` declared property wrappers. All other property wrapper traits also remain unchanged from SE-0258.
+
+## Detailed design
+
+Here are three examples of how a `let` wrapped property can make current iterations more effortless.
+
+The `Clamping` property wrapper from [SE-0258 Property Wrappers](https://github.com/apple/swift-evolution/blob/main/proposals/0258-property-wrappers.md#clamping-a-value-within-bounds) can be rewritten to use `let` as its storage and implementation do not change, no matter its application.
```

Comment:
What does the second half of this sentence mean? I feel like some words might be missing. How could the _implementation_ of the `Clamping` property wrapper change depending on its application?